test/ATokenFlexVotingFork.t.sol

```diff
@@ -1510,4 +1581,478 @@ contract VoteTest is AaveAtokenForkTest {
+      aToken.exposed_RawBalanceOf(_who)
+    );
+    assertEq(aToken.getPastStoredBalance(_who, block.number - 1), aToken.exposed_RawBalanceOf(_who));
+    assertGt(
+      _rawBalances[1], // The raw balance pre-withdrawal.
+      aToken.getPastStoredBalance(_who, block.number - _blocksJumpedThirdTime)
+    );
+  }
+
+  function test_GetPastStoredBalanceHandlesTransfers() public {
+    _initiateRebasing();
+
+    address _userA = makeAddr("GetPastStoredBalanceHandlesTransfers _userA");
+    address _userB = makeAddr("GetPastStoredBalanceHandlesTransfers _userB");
+    uint256 _amount = 4242 ether;
+
+    // Deposit.
+    _mintGovAndSupplyToAave(_userA, _amount);
+    uint256 _initRawBalanceUserA = aToken.exposed_RawBalanceOf(_userA);
+
+    // Advance the clock so that we checkpoint and let some rebasing happen.
+    vm.roll(block.number + 100);
+    vm.warp(block.timestamp + 100 days);
+
+    // Get the rebased balance for userA.
+    uint256 _initBalanceUserA = aToken.balanceOf(_userA);
+    uint256 _initBalanceUserB = aToken.balanceOf(_userB);
+    assertEq(_initBalanceUserB, 0);
+
+    // Transfer aTokens to userB.
+    vm.prank(_userA);
+    aToken.transfer(_userB, _initBalanceUserA / 3);
+    assertApproxEqAbs(aToken.balanceOf(_userA), 2 * _initBalanceUserA / 3, 1);
+    assertApproxEqAbs(aToken.balanceOf(_userB), _initBalanceUserA / 3, 1);
+
+    // Advance the clock so that we checkpoint.
+    vm.roll(block.number + 1);
+    vm.warp(block.timestamp + 1 days);
+
+    // Confirm voting weight has shifted.
+    assertApproxEqAbs(
+      aToken.getPastStoredBalance(_userA, block.number - 1),
+      2 * _initRawBalanceUserA / 3,
+      1 // In case of rounding.
+    );
+    assertApproxEqAbs(
+      aToken.getPastStoredBalance(_userB, block.number - 1),
+      _initRawBalanceUserA / 3,
+      1 // In case of rounding.
+    );
+  }
+
+  function test_GetPastStoredBalanceHandlesTransferFrom() public {
+    _initiateRebasing();
+
+    address _userA = makeAddr("GetPastStoredBalanceHandlesTransfers _userA");
+    address _userB = makeAddr("GetPastStoredBalanceHandlesTransfers _userB");
+    uint256 _amount = 4242 ether;
+
+    // Deposit.
+    _mintGovAndSupplyToAave(_userA, _amount);
+    uint256 _initRawBalanceUserA = aToken.exposed_RawBalanceOf(_userA);
+
+    // Advance the clock so that we checkpoint and let some rebasing happen.
+    vm.roll(block.number + 100);
+    vm.warp(block.timestamp + 100 days);
+
+    // Get the rebased balance for userA.
+    uint256 _initBalanceUserA = aToken.balanceOf(_userA);
+    uint256 _initBalanceUserB = aToken.balanceOf(_userB);
+    assertEq(_initBalanceUserB, 0);
+
+    // Transfer aTokens to userB.
+    vm.prank(_userA);
+    aToken.approve(address(this), type(uint256).max);
+    aToken.transferFrom(_userA, _userB, _initBalanceUserA / 3);
+    assertApproxEqAbs(aToken.balanceOf(_userA), 2 * _initBalanceUserA / 3, 1);
+    assertApproxEqAbs(aToken.balanceOf(_userB), _initBalanceUserA / 3, 1);
```

Comment:
Can we use `assertEq` here? We precisely know all numbers involved so should be no rounding errors

Reply:
We could, but I think this is easier to understand as a reader of the code

Reply:
This makes it look like we expect rounding errors though, but we don't

Reply:
Oh oh, yeah, I see what you mean. Gone 92d97c9